### PR TITLE
Back-porting PRs to 0.8.4 branch

### PR DESCRIPTION
**Pull request by j-wags**
We're aiming to make a 0.8.4 release simultaneous with 0.9.1. If possible, these releases will be feature-equivalent. 

0.8.4 will be the last release of the `openforcefield` namespace. 

This PR cherry picks the development that has occurred on `master` to the `0.8.4-rc` branch. It's my first time doing a cherry pick of this scale, so I suspect that this will require a little polishing to get CI happy again. 

- [x] #790 
- [x] #793 
- [x] #799 
- [x] #802 
- [x] #813
- [x] #816 
- [x] #827 
- [x] #828 
- [x] #831 
- [x] #832 
- [x] #839 
- [x] #849

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ on:
 
 jobs:
   test:
-    if: (github.event_name == 'schedule' && github.repository == 'openforcefield/openforcefield') || (github.event_name != 'schedule')
+    if: (github.event_name == 'schedule' && github.repository == 'openforcefield/openff-toolkit') || (github.event_name != 'schedule')
```
Comment (mattwthompson) [inline .github/workflows/CI.yml:20]: You probably already came across this, but CI will only run with an update to the `master` branch as it only runs when PRs are opened against `master` or `maintenance/.+` branches

**Comment by mattwthompson** [comment]
I have a few failing tests from the start
* Some of the links in the README pointed to the `master` branch, which is now broken because permalinks do not redirect to respect the new code paths. I updated these to point to the `084-update` branch.
* `z_3_hydroxy_propenal.sdf` wasn't in the tree, so I added it
* Some number of the recent PRs were missing (at least #831, probably also a WBO change). These are my currently failing tests
  * `openforcefield/tests/test_molecule.py::TestMolecule::test_apply_elf_conformer_selection`
    * (missing method in toolkit wrappers? or did not mean to add in test?)
  * `openforcefield/tests/test_toolkits.py::TestOpenEyeToolkitWrapper::test_assign_fractional_bond_orders` (several combinations of ELF10 methods and molecules)
  * `openforcefield/tests/test_toolkits.py::TestOpenEyeToolkitWrapper::test_assign_fractional_bond_orders_neutral_charge_mol`
  * `openforcefield/tests/test_toolkits.py::TestOpenEyeToolkitWrapper::test_assign_fractional_bond_orders_multi_conf`

This was done with my current development environment with both OpenEye toolkits and The RDKit installed. I also ran without OpenEye toolkits and everything passed - effectively a false positive because of

```
@requires_openeye
class TestMolecule:
```

Running with The RDKit but no OpenEye toolkits produces the same failed tests as above.

The current deprecation warning seems accurate. Would it be worth it to also link to https://github.com/openforcefield/openff-toolkit/issues/819, here or in the release notes?

```shell
__main__:1: DeprecationWarning: Importing this package as `import openforcefield.XXX` and `from openforcefield import XXX` was marked for deprecation in version `0.8.3`. From version `0.9.0` onwards this package will need to be imported as `import openff.toolkit.XXX` and `from openff.toolkit import XXX`. See the `0.8.3` release notes for more information.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (j-wags) [reply .github/workflows/CI.yml]: Thanks -- I'll try changing this line to see if I can get CI kicking again.

**Comment by j-wags** [comment]
Thanks, Matt. I'm going to keep banging on this today and will ping you when it's ready for review!

**Comment by j-wags** [comment]
Don't review this yet. I think there's actually a significant test failure. https://github.com/openforcefield/openff-toolkit/runs/2036523162#step:12:50065

This is acode adjacent to some of our current WBO work/bugfixes. I'm not sure that it's good to keep this as xfail. https://github.com/openforcefield/openff-toolkit/blame/master/openff/toolkit/tests/test_toolkits.py#L2930

I'll debug more tomorrow.

**Review by mattwthompson — APPROVED** [review]
Took a brief skim through the files and couldn't find anything objectionable. I'm restarting the tests; if the only failure is the stochastic one we've already discussed, LGTM

**Comment by j-wags** [comment]
Good news❓ Nothing fails any more❓❓ Hooray❓❓❓

**Pull request merged**